crates/bevy_internal/src/default_plugins.rs

```diff
@@ -1,144 +1,69 @@
-pub struct DefaultPlugins;
-
-impl PluginGroup for DefaultPlugins {
-    fn build(self) -> PluginGroupBuilder {
-        let mut group = PluginGroupBuilder::start::<Self>();
-        group = group
-            .add(bevy_log::LogPlugin::default())
-            .add(bevy_core::TaskPoolPlugin::default())
-            .add(bevy_core::TypeRegistrationPlugin)
-            .add(bevy_core::FrameCountPlugin)
-            .add(bevy_time::TimePlugin)
-            .add(bevy_transform::TransformPlugin)
-            .add(bevy_hierarchy::HierarchyPlugin)
-            .add(bevy_diagnostic::DiagnosticsPlugin)
-            .add(bevy_input::InputPlugin)
-            .add(bevy_window::WindowPlugin::default())
-            .add(bevy_a11y::AccessibilityPlugin);
-
+#![allow(dead_code)]
+use bevy_app::{plugin_group, Plugin, PluginGroup, PluginGroupBuilder};
+
+plugin_group!(
+    /// This plugin group will add all the default plugins for a *Bevy* application:
+    DefaultPlugins {
+        bevy_log:::LogPlugin,
+        bevy_core:::TaskPoolPlugin,
+        bevy_core:::TypeRegistrationPlugin,
+        bevy_core:::FrameCountPlugin,
+        bevy_time:::TimePlugin,
+        bevy_transform:::TransformPlugin,
+        bevy_hierarchy:::HierarchyPlugin,
+        bevy_diagnostic:::DiagnosticsPlugin,
+        bevy_input:::InputPlugin,
+        bevy_window:::WindowPlugin,
+        bevy_a11y:::AccessibilityPlugin,
         #[cfg(feature = "bevy_asset")]
-        {
-            group = group.add(bevy_asset::AssetPlugin::default());
-        }
-
+        bevy_asset:::AssetPlugin,
         #[cfg(feature = "bevy_scene")]
-        {
-            group = group.add(bevy_scene::ScenePlugin);
-        }
-
+        bevy_scene:::ScenePlugin,
         #[cfg(feature = "bevy_winit")]
-        {
-            group = group.add(bevy_winit::WinitPlugin::default());
-        }
-
+        bevy_winit:::WinitPlugin,
         #[cfg(feature = "bevy_render")]
-        {
-            group = group
-                .add(bevy_render::RenderPlugin::default())
-                // NOTE: Load this after renderer initialization so that it knows about the supported
-                // compressed texture formats
-                .add(bevy_render::texture::ImagePlugin::default());
-
-            #[cfg(all(not(target_arch = "wasm32"), feature = "multi-threaded"))]
-            {
-                group = group.add(bevy_render::pipelined_rendering::PipelinedRenderingPlugin);
-            }
-        }
-
+        bevy_render:::RenderPlugin,
+        // NOTE: Load this after renderer initialization so that it knows about the supported
+        // compressed texture formats
+        #[cfg(feature = "bevy_render")]
+        bevy_render::texture:::ImagePlugin,
+        #[cfg(feature = "bevy_render")]
+        #[custom(cfg(all(not(target_arch = "wasm32"), feature = "multi-threaded")))]
+        bevy_render::pipelined_rendering:::PipelinedRenderingPlugin,
```

Comment:
It would be nice if it would document not targeting `wasm32`

Reply:
It would be nice, but it would be quite difficult to implement this with the macro.

Reply:
Besides, even the original `DefaultPlugins` documentation didn't mention this requirement.

Reply:
It did.
https://github.com/bevyengine/bevy/blob/10f95956a6c4e36f00a1749434c5a87933d95bb2/crates/bevy_internal/src/default_plugins.rs#L19

Reply:
Ah mb